nl_server/routes.py

```diff
@@ -117,36 +97,38 @@ def detect_verbs():
   List[str]
   """
   query = str(escape(request.args.get('q')))
-  nl_model = current_app.config[config.REGISTRY_KEY].attribute_model()
-  return json.dumps(nl_model.detect_verbs(query.strip()))
+  r: Registry = current_app.config[REGISTRY_KEY]
+  return json.dumps(r.get_attribute_model().detect_verbs(query.strip()))
 
 
-@bp.route('/api/embeddings_version_map/', methods=['GET'])
+@bp.route('/api/server_config/', methods=['GET'])
 def embeddings_version_map():
-  return json.dumps(asdict(current_app.config[config.CATALOG_KEY]))
+  r: Registry = current_app.config[REGISTRY_KEY]
+  server_config = r.server_config()
+  return json.dumps(asdict(server_config))
 
 
 @bp.route('/api/load/', methods=['GET'])
 def load():
-  loader.load_custom_embeddings(current_app)
-  return json.dumps(asdict(current_app.config[config.CATALOG_KEY]))
-
-
-def _get_indexes(registry: Registry, idx: str) -> List[Embeddings]:
-  nl_embeddings: List[Embeddings] = []
-
-  if is_custom_dc() and idx != config.CUSTOM_DC_INDEX:
-    # Order custom index first, so that when the score is the same
-    # Custom DC will be preferred.
-    emb = registry.get_index(config.CUSTOM_DC_INDEX)
-    if emb:
-      nl_embeddings.append(emb)
-
-  emb = registry.get_index(idx)
-  if emb:
-    nl_embeddings.append(emb)
-
-  return nl_embeddings
+  try:
+    current_app.config[REGISTRY_KEY] = registry.build()
+  except Exception as e:
+    logging.error(f'Server registry not built due to error: {str(e)}')
+  r: Registry = current_app.config[REGISTRY_KEY]
+  server_config = r.server_config()
+  return json.dumps(asdict(server_config))
+
+
+def _get_indexes(r: Registry, idx_types: List[str]) -> List[Embeddings]:
+  embeddings: List[Embeddings] = []
+  for idx in idx_types:
+    try:
+      emb = r.get_index(idx)
+      if emb:
+        embeddings.append(emb)
+    except Exception as e:
+      logging.warning(f'Failed to load index {idx}: {e}')
```

Comment:
Why do we need this try-except?  get_index seems to do a `.get()` on the dict?

Reply:
seems to be a wrong change. reverted back.